core/domain/learner_playlist_services_test.py

```diff
@@ -100,162 +101,128 @@ def _get_all_learner_playlist_collection_ids(self, user_id):
+            self.user_id, 'SUBSCRIBED_ID')
 
-    def test_mark_collection_to_be_played_later(self):
+        # Now if we try to add the same collection, it shouldn't be added.
+        learner_progress_services.add_collection_to_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
         self.assertEqual(
             self._get_all_learner_playlist_collection_ids(self.user_id), [])
 
-        # Add a collection to the learner playlist.
+        learner_playlist_services.remove_collection_from_learner_playlist(
+            self.user_id, 'SUBSCRIBED_ID')
+
+        # Test adding the collection_id if it is already in
+        # learner_playlist.collection_ids
+        # Add two collections to the learner playlist.
         learner_progress_services.add_collection_to_learner_playlist(
             self.user_id, self.COL_ID_0)
-        self.assertEqual(
-            self._get_all_learner_playlist_collection_ids(
-                self.user_id), [self.COL_ID_0])
-
-        # Add another collection.
         learner_progress_services.add_collection_to_learner_playlist(
             self.user_id, self.COL_ID_1)
         self.assertEqual(
             self._get_all_learner_playlist_collection_ids(
                 self.user_id), [self.COL_ID_0, self.COL_ID_1])
 
-        # Add the first collection in a different position.
+        # Add the first collection to the second position.
         learner_progress_services.add_collection_to_learner_playlist(
             self.user_id, self.COL_ID_0,
             position_to_be_inserted=1)
         self.assertEqual(
             self._get_all_learner_playlist_collection_ids(
                 self.user_id), [self.COL_ID_1, self.COL_ID_0])
 
-        # Add the first collection in the first position.
-        learner_progress_services.add_collection_to_learner_playlist(
-            self.user_id, self.COL_ID_0,
-            position_to_be_inserted=0)
-        self.assertEqual(
-            self._get_all_learner_playlist_collection_ids(
-                self.user_id), [self.COL_ID_0, self.COL_ID_1])
-
-        # Add a collection to the in progress list of the user.
-        learner_progress_services.mark_collection_as_incomplete(
-            self.user_id, self.COL_ID_3)
-
-        # Test that the collection added to the in progress list doesn't get
-        # added to the learner playlist.
-        learner_progress_services.add_collection_to_learner_playlist(
-            self.user_id, self.COL_ID_3)
-        self.assertEqual(
-            self._get_all_learner_playlist_collection_ids(
-                self.user_id), [self.COL_ID_0, self.COL_ID_1])
-
-        # Empty the learner playlist.
-        learner_playlist_services.remove_collection_from_learner_playlist(
-            self.user_id, self.COL_ID_0)
-        learner_playlist_services.remove_collection_from_learner_playlist(
-            self.user_id, self.COL_ID_1)
+    def test_number_of_collections_cannot_exceed_max(self):
```

Comment:
Split it into two test (same as the corresponding test for exploration)

Reply:
I'll do whatever I do to explorations!